packages/extension/src/background/RequestBytesSign.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright 2019 @polkadot/extension authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { KeyringPair } from '@polkadot/keyring/types';
+import { RequestSign } from './types';
+import { SignerPayloadJSON, SignerPayloadRaw } from '@polkadot/types/types';
+import { u8aToHex, stringToU8a } from '@polkadot/util';
+
+export default class RequestBytesSign implements RequestSign {
+  inner: SignerPayloadJSON | SignerPayloadRaw;
+
+  constructor (inner: SignerPayloadRaw) {
+    this.inner = inner;
+  }
+
+  sign (pair: KeyringPair): { signature: string } {
+    const inner = this.inner as SignerPayloadRaw;
+    const signedBytes = pair.sign(stringToU8a(inner.data));
```

Comment:
According to the `SignerPayloadRawBase` signature, 

```typescript
export interface SignerPayloadRawBase {
    /**
     * @description The hex-encoded data for this request
     */
    data: string;
    /**
     * @description The type of the contained data
     */
    type?: 'bytes' | 'payload';
}
```

`data` will always be hex, so it is up to the caller to pass valid data.

Also, `type?: 'bytes' | 'payload';` doesn't make sense. One can simply `const bytes = extrinsic.toU8a(true)` and `signRaw(bytes)`. Can we remove it?